src/entity/object/FallingBlock.php

```diff
@@ -128,14 +129,19 @@ protected function entityBaseTick(int $tickDiff = 1) : bool{
 			if($this->onGround || $blockTarget !== null){
 				$this->flagForDespawn();
 
+				$blockResult = $blockTarget ?? $this->block;
 				$block = $world->getBlock($pos);
 				if(!$block->canBeReplaced() || !$world->isInWorld($pos->getFloorX(), $pos->getFloorY(), $pos->getFloorZ()) || ($this->onGround && abs($this->location->y - $this->location->getFloorY()) > 0.001)){
 					$world->dropItem($this->location, $this->block->asItem());
+					$world->addSound($pos->add(0.5, 0.5, 0.5), new BlockBreakSound($blockResult));
 				}else{
-					$ev = new EntityBlockChangeEvent($this, $block, $blockTarget ?? $this->block);
+					$ev = new EntityBlockChangeEvent($this, $block, $blockResult);
 					$ev->call();
 					if(!$ev->isCancelled()){
 						$world->setBlock($pos, $ev->getTo());
+						if($this->onGround && $blockResult instanceof Fallable && ($sound = $blockResult->getFallSound()) !== null){
```

Comment:
Probably it would make more sense to use `$ev->getTo()` for the result, so that the correct sounds are played if the block is changed by a plugin.

Reply:
done